src/ray/core_worker/core_worker.cc

```diff
@@ -1548,7 +1555,7 @@ std::vector<rpc::ObjectReference> CoreWorker::SubmitTask(
 
 Status CoreWorker::CreateActor(const RayFunction &function,
                                const std::vector<std::unique_ptr<TaskArg>> &args,
-                               const ActorCreationOptions &actor_creation_options,
+                               ActorCreationOptions actor_creation_options,
```

Comment:
Why do you change this? I believe the `actor_creation_options` is not passed anywhere else. You can use a local variable (without `std::shared_ptr`) to store the computed `is_detached` value.